stats-functions/hourly_edge_requests/src/main.py

```diff
@@ -0,0 +1,258 @@
+import os
+import argparse
+import logging
+from google.cloud import logging as gc_logging
+
+from datetime import datetime, timedelta, timezone
+from dateutil import parser
+
+import functions_framework
+from cloudevents.http import CloudEvent
+from google.cloud.sql.connector import Connector, IPTypes
+import pymysql
+
+import fastly
+from fastly.api import stats_api
+from fastly.model.stats import Stats
+
+from sqlalchemy import create_engine, delete
+from sqlalchemy.engine.base import Engine
+from sqlalchemy.orm import sessionmaker
+
+from stats_entities.site_usage import HourlyRequests
+from models import Database, FastlyStatsApiResponse
+from pydantic import ValidationError
+
+logger = logging.getLogger(__name__)
+
+
+class NoRetryError(Exception):
+    pass
+
+
+class HourlyEdgeRequestsJob:
+    MAX_EVENT_AGE_IN_MINUTES = 50  # to limit retries
+    FASTLY_SERVICE_ID = {"arxiv.org": "umpGzwE2hXfa2aRXsOQXZ4"}
+    FASTLY_NODE_NUMBER = 0  # existing convention, corresponds to 'fastly'
+
+    def __init__(self):
+        self.hour_delay = os.getenv("HOUR_DELAY", 1)
+
+        self.env = os.getenv("ENV")
+        self.write_db = Database(
+            os.getenv("WRITE_DB_INSTANCE"),
+            os.getenv("WRITE_DB_USER"),
+            os.getenv("WRITE_DB_PW"),
+            os.getenv("WRITE_DB_NAME"),
+        )
+        self.project = "arxiv-production" if self.env == "PROD" else "arxiv-development"
+        self.ip_type = (
+            IPTypes.PRIVATE if os.environ.get("PRIVATE_IP") else IPTypes.PUBLIC
+        )
+        self.log_level = os.getenv("LOG_LEVEL", "INFO")
+        self.log_locally = os.getenv("LOG_LOCALLY", False)
```

Comment:
Will code like this result in self.log_locally set to the string `"false"`  or `"0"` which resolves as the bool `True`?   
I think in other places we use pydantic settings to make this go smoother https://docs.pydantic.dev/latest/concepts/pydantic_settings/